src/Forms/GridField/GridFieldConfig.php

```diff
@@ -91,12 +91,19 @@ public function removeComponent(GridFieldComponent $component)
     }
 
     /**
-     * @param string $type Class name or interface
+     * @param string|string[] $types Class name or interface, or an array of the same
      * @return $this
      */
-    public function removeComponentsByType($type)
+    public function removeComponentsByType($types)
     {
-        $components = $this->getComponentsByType($type);
+        if (is_array($types)) {
```

Comment:
One code path to do one task please... just convert the string to a single-item array and then we don't duplicate logic. :)